ankerctl.py

```diff
@@ -81,6 +82,9 @@ def main(ctx, verbose, quiet, insecure):
 
     env.upgrade_config_if_needed()
 
+    env.printer = printer
```

Comment:
We should add a method to the `Environment` class to handle this, instead of patching the value into it: `def select_printer(self, index)` on `Environment`.

This function should do the following:
 1. Check that the index is within range
 2. Open the config, and sort the list in a well-defined way (see later comment on `enumerate`)
 3. Save this list to `self._printers`
 4. Save the index to `self._index`

Then, we can add a property:

```python
@property
def printer(self):
    return self._printers[self._index]
```

This way, we can use `env.printer` to refer to the selected printer, anywhere we need it. This should make the whole thing simpler.